nixos/doc/manual/development/option-types.section.md

```diff
@@ -400,44 +400,55 @@ Composed types are types that take a type as parameter. `listOf
 
 `types.attrsOf` *`t`*
 
-:   An attribute set of where all the values are of *`t`* type. Multiple
-    definitions result in the joined attribute set.
+:   _Shorthand for `types.attrsWith { lazy = false; elemType = t; }`._
 
-    ::: {.note}
-    This type is *strict* in its values, which in turn means attributes
-    cannot depend on other attributes. See `
-           types.lazyAttrsOf` for a lazy version.
-    :::
+    An attribute set of where all the values are of *`t`* type.
+
+    This type is *strict* in its values, which in turn means attributes cannot depend on other attributes.
+    See `lazy` in `attrsWith`.
 
 `types.lazyAttrsOf` *`t`*
 
-:   An attribute set of where all the values are of *`t`* type. Multiple
-    definitions result in the joined attribute set. This is the lazy
-    version of `types.attrsOf
-          `, allowing attributes to depend on each other.
+:   _Shorthand for `types.attrsWith { lazy = true; elemType = t; }`._
 
-    ::: {.warning}
-    This version does not fully support conditional definitions! With an
-    option `foo` of this type and a definition
-    `foo.attr = lib.mkIf false 10`, evaluating `foo ? attr` will return
```

Comment:
That example is actually valuable to illustrate what is meant by "more attributes than desirable". Instead of a warning, how about putting that text into an example box? Although it's not entirely wrong to make it a warning because it's a potential pitfall and source of inscrutable errors. May be enough to move it to the documentation of `lazy`.

Reply:
The `.note` in `attrsOf` leads to a similar amount of inscrutability.

I agree it wasn't wrong. I just think it doesn't make much sense anymore, now that it's just an explanation of the semantics.

Reply:
It's an *illustration*, and I'd appreciate if we treated it like an example, which is always very good to have. I don't mind where it lives. Also the piece on `emptyValue` would be gone entirely. Admittedly it's a niche use case, but this reference documentation, the level of detail needs no ceiling here.